### PR TITLE
Update submodule versions.

### PR DESCRIPTION
**Pull request by brianhuffman**

### CODE REVIEW

File: cabal.GHC-8.10.3.config [hunk 1]
```diff
@@ -195,7 +195,7 @@ constraints: any.Cabal ==3.2.1.0,
              any.logict ==0.7.0.3,
              any.math-functions ==0.3.4.1,
              math-functions +system-erf +system-expm1,
-             any.megaparsec ==9.0.1,
+             any.megaparsec ==8.0.0,
```
Comment (RyanGlScott) [inline cabal.GHC-8.10.3.config:198]: What caused the `megaparsec` version to decrease?

Comment (brianhuffman) [reply cabal.GHC-8.10.3.config:198]: It was a constraint in the `crucible-syntax` package (see commit message above for 650958b).

Comment (brianhuffman) [reply cabal.GHC-8.10.3.config:198]: Probably the `crucible-syntax` upper bound can be safely bumped (damn cabal for not distinguishing between "known not to work" upper bounds and "not known to work") but there are other later crucible PRs that are already in crucible master and that require further saw-script changes (i.e. GaloisInc/crucible#735 and #1290).

Comment (RyanGlScott) [reply cabal.GHC-8.10.3.config:198]: Good to know, thanks. Indeed, there's no need to hold up this MR for this—we can always bump the `megaparsec` version bounds separately.

**Review by RyanGlScott — APPROVED** [review]

File: cabal.GHC-8.10.3.config (hunk 1 above)
Comment (RyanGlScott) [reply cabal.GHC-8.10.3.config:198]: I've opened GaloisInc/crucible#743 for this.

**Pull request merged**